lib/contentTypeParser.js

```diff
@@ -393,10 +393,16 @@ function ParserListItem (contentType) {
   // we pre-calculate all the needed information
   // before content-type comparsion
   const parsed = safeParseContentType(contentType)
-  this.type = parsed.type
+  this.isEssence = contentType.indexOf(';') === -1
+  // we should not allow empty string for parser list item
+  // because it would becomes a match-all handler
+  if (this.isEssence === false && parsed.type === '') {
+    this.type = contentType.split(';')[0]
```

Comment:
I thought very long about this. I am not convinced that it will avoid the case that we could replace the wildcard.

E.g. contentType with the value ";" maybe also results being a wildcard

Reply:
Or am i having a brainfart?